otx/core/base_dataset_adapter.py

```diff
@@ -0,0 +1,225 @@
+            test_data_roots (str): Path for test data
+            test_ann_files (str): Path for test annotation data
+            unlabeled_data_roots (str): Path for unlabeled data 
+            unlabeled_file_lists (str): Path for unlabeled file list
+        
+        Returns:
+            DatumaroDataset: Datumaro Dataset
+        """
+        # Find self.data_type and task_type
+        data_type_candidates = self._detect_dataset_format(path=train_data_roots)
+        logger.info('[*] Data type candidates: {}'.format(data_type_candidates))
+        self.data_type = self._select_data_type(data_type_candidates) 
+        logger.info('[*] Selected data type: {}'.format(self.data_type))
+
+        # Construct dataset for training, validation, unlabeled
+        self.dataset = {}
+        logger.info('[*] Importing Datasets...')
+        datumaro_dataset = DatumaroDataset.import_from(train_data_roots, format=self.data_type)
+
+        # Prepare subsets by using Datumaro dataset
+        for k, v in datumaro_dataset.subsets().items():
+            if 'train' in k or 'default' in k:
+                self.dataset[Subset.TRAINING] = v
+            elif 'val' in k:
+                self.dataset[Subset.VALIDATION] = v
+        
+        # If validation is manually defined --> set the validation data according to user's input
+        if val_data_roots is not None:
+            val_data_candidates = self._detect_dataset_format(path=val_data_roots)
+            val_data_type = self._select_data_type(val_data_candidates)
+            assert self.data_type == val_data_type, "The data types of training and validation must be same, the type of train:{} val:{}".format(
+               self.data_type, val_data_type 
+            )
+            self.dataset[Subset.VALIDATION] = DatumaroDataset.import_from(val_data_roots, format=val_data_type)
+
+        if Subset.VALIDATION not in self.dataset.keys():
+            #TODO: auto_split
+            pass
+
+        if unlabeled_data_roots is not None:
+            self.dataset[Subset.UNLABELED] = DatumaroDataset.import_from(unlabeled_data_roots, format='image_dir')
+        
+        return self.dataset
+
+    @abstractmethod
+    def convert_to_otx_format(self, datumaro_dataset: dict) -> DatasetEntity:
+        """Convert DatumaroDataset to the DatasetEntity.
+        Args:
+            datumaro_dataset (dict): A Dictionary that includes subset dataset(DatasetEntity) for training/validation/test
+        Returns:
+            DatasetEntity: 
+        """
+        pass
+
+    def _detect_dataset_format(self, path: str) -> str:
+        """ Detect dataset format (ImageNet, COCO, ...). """
+        return datumaro.Environment().detect_dataset(path=path)
+
+    def _generate_empty_label_entity(self) -> LabelGroup:
+        """ Generate Empty Label Group for H-label, Multi-label Classification. """
+        empty_label = LabelEntity(name="Empty label", is_empty=True, domain=self.domain)
```

Comment:
I am not sure that "Empty label" is okay. It is known that using space with lxml yields error. Please see https://jira.devtools.intel.com/browse/CVS-97188.